cfme/intelligence/reports/saved.py

```diff
@@ -1,62 +1,91 @@
+            self.saved_reports.is_opened and
+            self.saved_reports.tree.currently_selected == ["All Saved Reports"] and
+            self.title.text == "All Saved Reports"
+        )
+
+
+class SavedReportDetailsView(CustomSavedReportDetailsView):
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_intel_reports and
+            self.saved_reports.is_opened and
+            self.saved_reports.tree.currently_selected == ([
+                "All Reports",
+                self.context["object"].name,
+                self.context["object"].run_at_datetime
+            ]) and
+            self.title.text == 'Saved Report "{}"'.format(self.context["object"].name)
+        )
+
+
+class SavedReportView(AllSavedReportsView):
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_intel_reports and
+            self.saved_reports.is_opened and
+            self.saved_reports.tree.currently_selected == [
+                "All Saved Reports",
+                self.context["object"].name
+            ] and
+            self.title.text == 'Saved Report "{}"'.format(self.context["object"].name)
+        )
 
 
 class SavedReport(Navigatable):
 
-    def __init__(self, name, timestamp, appliance=None):
+    def __init__(self, name, run_at_datetime, queued_datetime_in_title, appliance=None):
         Navigatable.__init__(self, appliance)
         self.name = name
-        self.timestamp = timestamp
-
-    def delete(self, cancel):
-        navigate_to(self, 'Delete')
-        sel.handle_alert(cancel)
-
-    @classmethod
-    def get_all_saved_reports_by_name(cls, name):
-        navigate_to(cls, 'All')
-        accordion.tree("Saved Reports", "All Saved Reports", name)
-        dates = []
-        try:
-            for row in reports_table.rows():
-                dates.append(sel.text(row.queued_at).encode("utf-8").strip())
-        except sel.NoSuchElementException:
-            pass
-        return dates
-
-    def go_to_latest_saved_report(self):
-        latest = self.get_all_saved_reports_by_name(self.name)[0]
-        navigate_to(self, 'Details')
-        reports_table.click_cell("queued_at", latest)
-
-
-@navigator.register(SavedReport, 'All')
-class SavedReportAll(CFMENavigateStep):
-    prerequisite = NavigateToObject(Report, 'SavedReports')
-
-
-@navigator.register(SavedReport, 'Details')
+        self.run_at_datetime = run_at_datetime
+        self.queued_datetime_in_title = queued_datetime_in_title
+
+    def delete(self, cancel=False):
+        view = navigate_to(self, "Details")
+        view.configuration.item_select(
+            "Delete this Saved Report from the Database",
+            handle_alert=not cancel
+        )
+        if cancel:
+            assert view.is_displayed
+            view.flash.assert_no_error()
+        else:
+            view.flash.assert_no_error()
+            view.flash.assert_message("Successfully deleted Saved Report from the CFME Database")
```

Comment:
OPTIONAL: There is `assert_success_message` method and it contains both of `assert_no_error`  and `assert_message`